### PR TITLE
set allow_unicode to True where we use slugify

### PR DESCRIPTION
**Pull request by gumaerc**
### Screenshots (if appropriate):

- [ ] Desktop screenshots
- [ ] Mobile width screenshots

### CODE REVIEW

**Comment by pdpinch** [comment]
What slug/URL do we get when the title is set to テスト ページ ?

File: websites/models.py [hunk 1]
```diff
@@ -256,11 +256,11 @@ def url_path_from_metadata(self, metadata: dict | None = None):
                     content = self.websitecontent_set.filter(type=section_type).first()
                     if content:
                         value = get_dict_field(content.metadata, section_field)
-                if not value:  # noqa: SIM108
+                if not value:
                     # Incomplete metadata required for url
                     value = section
                 else:
-                    value = slugify(value.replace(".", "-"))
+                    value = slugify(value.replace(".", "-"), allow_unicode=True)
```
Comment (HussainTaj-arbisoft) [inline websites/models.py:263]: The Studio UI doesn't allow Unicode characters in the url_path. I wonder if we should keep `allow_unicode=False` here since Unicode characters are invalid for this field.

Screenshot 2024-02-07 at 5 13 45 PM

File: websites/models_test.py [hunk 2]
```diff
@@ -283,6 +283,28 @@ def test_website_url_path_from_metadata(missing_keys):
     assert website.url_path_from_metadata(metadata) == expected_url_path
 
 
+def test_website_url_path_from_metadata_unicode():
+    """None should be returned for a site without a starter"""
```
Comment (HussainTaj-arbisoft) [inline websites/models_test.py:287]:
```suggestion
    """url_path_from_metadata should correctly generate a url/format for unicode input."""
```

File: websites/models_test.py [hunk 3]
```diff
@@ -283,6 +283,28 @@ def test_website_url_path_from_metadata(missing_keys):
...
+    starter = WebsiteStarterFactory.create(
+        config={
+            "site-url-format": "[meta:course_nr]-[meta:title]-[meta:term]-[meta:year]"
+        }
+    )
+    website = WebsiteFactory.create(starter=starter, not_published=True)
+    course_nr = "1-1" if metadata.get("course_nr") is not None else "[meta:course_nr]"
+    title = "テスト-ウェブサイト" if metadata.get("title") is not None else "[meta:title]"
+    term = "fall" if metadata.get("term") is not None else "[meta:term]"
+    year = metadata.get("year", "[meta:year]")
```
Comment (HussainTaj-arbisoft) [inline websites/models_test.py:303]: Unless you meant to parameterize `metadata`, this can be simplified.

```suggestion
    course_nr = "1-1"
    title = "テスト-ウェブサイト"
    term = "fall"
    year = "2025"
```

File: websites/views.py [hunk 4]
```diff
@@ -190,7 +190,7 @@ def get_serializer(self, *args, **kwargs):
             and self.request.data.get("title")
         ):
             request_data = self.request.data.copy()
-            request_data["name"] = slugify(request_data["title"])
+            request_data["name"] = slugify(request_data["title"], allow_unicode=True)
```
Comment (HussainTaj-arbisoft) [inline websites/views.py:193]: Unicode `name`s don't seem to work for me in Studio. Seems like a frontend issue, because I see no request being made to fetch data from the backend.

Screenshot 2024-02-07 at 5 26 49 PM

File: websites/views_test.py [hunk 5]
```diff
@@ -577,7 +577,27 @@ def test_websites_autogenerate_name(drf_client):
...
+def test_websites_autogenerate_name_unicode(drf_client):
+    """The autogenerated name should be compatible with unicode characters"""
+    superuser = UserFactory.create(is_superuser=True)
+    drf_client.force_login(superuser)
+    starter = WebsiteStarterFactory.create(
+        source=constants.STARTER_SOURCE_GITHUB,
+        status=constants.WebsiteStarterStatus.ACTIVE,
+    )
+    website_title = "テスト ページ"
+    website_short_id = "テスト_ページ"
```
Comment (HussainTaj-arbisoft) [inline websites/views_test.py:599]: It seems like GitHub doesn't allow unicode and our Studio UI forbits it as well for `short-id`. We should probably not test this here, and maybe add a test to ensure short-id is not unicode compatible.

Screenshot 2024-02-07 at 5 31 12 PM

**Review by HussainTaj-arbisoft — COMMENTED** [review]
The changes seem good. It solves the `filename` issue we had. I do have a few suggestions and comments. 

I think we can merge this change; however, supporting Unicode characters throughout Studio might require more thought.

---

@gumaerc I was unable to test the following functions locally with unicode names. We'll probably need to test them on RC:

- Video/caption workflow (a video with a special name). To test YouTube, MediaConvert, and 3Play integrations with unicode names.
- Search Index

---

This trivial, not urgent, issue with the `file` field showing encoded names.

Screenshot 2024-02-07 at 4 46 42 PM

**Comment by HussainTaj-arbisoft** [comment]
Perhaps we should make only the `filename`s Unicode compatible and leave all other fields incompatible for now.

**Comment by gumaerc** [comment]
> What slug/URL do we get when the title is set to テスト ページ ?

We would get a slug of `テスト-ページ`, which modern browsers do support but older operating systems / browsers may have issues with this. @HussainTaj-arbisoft has highlighted a number of areas where Unicode characters will be problematic, and I'm going to step back the scope of this PR to take that into account, although I didn't get a chance to address this today.

**Comment by gumaerc** [comment]
@HussainTaj-arbisoft This is ready for another look

**Review by HussainTaj-arbisoft — APPROVED** [review]
Looks good 👍 

A reminder that I'm unable to test the video workflow locally, so we should probably do that on RC. I don't expect any issues with it; it's just to be safe.
